packages/react-components/react-motion/library/src/factories/createPresenceComponent.ts

```diff
@@ -173,7 +176,16 @@ export function createPresenceComponent<MotionParams extends Record<string, Moti
         },
         // Excluding `isFirstMount` from deps to prevent re-triggering the animation on subsequent renders
         // eslint-disable-next-line react-hooks/exhaustive-deps
-        [animateAtoms, handleRef, isReducedMotion, handleMotionFinish, handleMotionStart, handleMotionCancel, visible],
+        [
+          animateAtoms,
+          handleRef,
+          isReducedMotion,
+          handleMotionFinish,
+          handleMotionStart,
+          handleMotionCancel,
+          visible,
+          disableMotions,
```

Comment:
Let's store `disableMotions` in `optionsRef`, so it won't trigger animation start if value changes